File: objc/ZendriveSDKDemo/Model/LocationPoint.m
//
//  LocationPoint.m
//  ZendriveSDKDemo
//
//  Created by Yogesh on 3/7/15.
//  Copyright (c) 2015 Zendrive. All rights reserved.
//

#import "LocationPoint.h"

#define kLatitudeKey    @"latitude"
#define kLongitudeKey   @"longitude"
@implementation LocationPoint

- (instancetype)initWithLatitude:(double)latitude longitude:(double)longitude
{
    self = [super init];
    if (self) {
        _latitude = latitude;
        _longitude = longitude;
    }
    return self;
}

- (instancetype)initWithDictionary:(NSDictionary *)dictionary
{
    self = [super init];
    if (self) {
        NSNumber *latitude = [dictionary objectForKey:kLatitudeKey];
        _latitude = latitude.doubleValue;

        NSNumber *longitude = [dictionary objectForKey:kLongitudeKey];
        _longitude = longitude.doubleValue;
    }
    return self;
}

- (NSDictionary *)toDictionary {
    return @{kLatitudeKey:@(self.latitude), kLongitudeKey:@(self.longitude)};
}
@end

File: swift/ZendriveSDKDemoSwift/Model/LocationPoint.swift
//
//  LocationPoint.swift
//  ZendriveSDKDemoSwift
//
//  Created by amrit on 23/07/19.
//  Copyright © 2019 zendrive. All rights reserved.
//

import Foundation

final class LocationPoint {
    private let latitudeKey = "latitude"
    private let longitudeKey = "longitude"
    private(set) var latitude = 0.0
    private(set) var longitude = 0.0
    init(latitude: Double, longitude: Double) {
        self.latitude = latitude
        self.longitude = longitude
    }

    init(dictionary: [AnyHashable : Any]?) {
        if let longitude = dictionary?[longitudeKey] as? Double, let latitude = dictionary?[latitudeKey] as? Double {
            self.longitude = longitude
            self.latitude = latitude
        } else {
            AppLogger.logInfo("No waypoints")
        }
    }

    func toDictionary() -> [AnyHashable : Any] {
        return [latitudeKey: self.latitude, longitudeKey: self.longitude]
    }
}
